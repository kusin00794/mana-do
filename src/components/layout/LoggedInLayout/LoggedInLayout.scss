@import '../../../styles/variables.scss';

nav {
  display: flex;
}

.navbar {
  position: sticky;
  top: 0;
  left: 250px;
  right: 0;
  background: $white-color;
  z-index: 1030;

  .nav {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    padding: 0 15px;

    &-item {
      margin-right: 10px;
    }

    &-text {
      color: $black-color;
    }
  }
}

.content {
  display: flex;
  flex-direction: row;
  max-width: 100%;
  padding: 1.5rem;
}
