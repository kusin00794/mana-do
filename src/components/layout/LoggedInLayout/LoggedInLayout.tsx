import React, { FC } from 'react';
import { Link, useHistory } from 'react-router-dom';

import SideBar from '../../sideBar/sideBar';
import Breadcrumb from '../../../components/Breadcrumb/Breadcrumb';

import RoutesString from '../../../pages/routesString';

import '../../../styles/common.scss';
import './LoggedInLayout.scss';

type UserInfo = {
  userId: string;
};

const LoggedInLayout: FC = ({ children }) => {
  const history = useHistory();
  const defaultUserInfo: UserInfo = {
    userId: 'User',
  };
  const auth = localStorage.getItem('auth');
  const userInfo = auth ? JSON.parse(auth) : defaultUserInfo;
  const userName = userInfo.userId ?? 'User';

  const handleLogout = (e: React.MouseEvent) => {
    e.preventDefault();
    localStorage.removeItem('auth');
    return history.push(RoutesString.Login);
  };

  return (
    <div className="layout logged-in-layout">
      <SideBar />
      <div className="wrap">
        <div className="navbar">
          <nav>
            <Breadcrumb />
            <ul className="nav ml-auto">
              <li className="nav-item">
                Welcome
                <strong className="ml-1">{userName}</strong>,
              </li>
              <li className="nav-item">
                <Link className="nav-text" to="/" onClick={handleLogout}>
                  <i className="fa fa-power-off" />
                </Link>
              </li>
            </ul>
          </nav>
        </div>
        <div className="content">{children}</div>
      </div>
    </div>
  );
};

export default LoggedInLayout;
