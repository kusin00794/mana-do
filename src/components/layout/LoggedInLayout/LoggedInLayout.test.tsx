import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import LoggedInLayout from './LoggedInLayout';
import { testSnapshots } from '../../../utils/test';

const Wrapper = () => (
  <BrowserRouter>
    <LoggedInLayout>
      <div>Mocked Component</div>
    </LoggedInLayout>
  </BrowserRouter>
);

describe('Snapshot for LoggedInLayout', () => {
  testSnapshots(Wrapper, [
    {
      props: {},
      description: 'render LoggedInLayout',
    },
  ]);
});
