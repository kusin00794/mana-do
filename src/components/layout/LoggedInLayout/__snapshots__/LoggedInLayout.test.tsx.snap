// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for LoggedInLayout snapshots render LoggedInLayout 1`] = `
<div
  className="layout logged-in-layout"
>
  <div
    className="sidebar-fluid"
  >
    <div
      className="sidebar"
    >
      <h3
        className="sidebar-title text-center"
      >
        Menu
      </h3>
      <div
        className="sidebar-section"
      >
        <ul
          className="sidebar-menu"
        >
          <li
            className="sidebar-menu-entry"
          >
            <a
              className="sidebar-menu-entry-link active"
              href="/"
              onClick={[Function]}
            >
              <span>
                Home
              </span>
            </a>
          </li>
          <li
            className="sidebar-menu-entry"
          >
            <a
              className="sidebar-menu-entry-link"
              href="/todo"
              onClick={[Function]}
            >
              <span>
                To Do
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="wrap"
  >
    <div
      className="navbar"
    >
      <nav>
        <ul
          className="nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-text"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-home"
              />
            </a>
          </li>
        </ul>
        <ul
          className="nav ml-auto"
        >
          <li
            className="nav-item"
          >
            Welcome
            <strong
              className="ml-1"
            >
              User
            </strong>
            ,
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-text"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fa fa-power-off"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="content"
    >
      <div>
        Mocked Component
      </div>
    </div>
  </div>
</div>
`;
